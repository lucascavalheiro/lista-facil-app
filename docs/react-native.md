# Getting Started with React Native

## Setting Up Your Machine for App Development

- iOS (You require a Mac to develop iOS Apps) Simply download XCode
- [Android on Mac](https://medium.com/pvtl/react-native-android-development-on-mac-ef7481f65e47#.w2hel88zy)
- Android on Windows - coming soon...

## Install React Native

[React Native - Get Started Guide](https://facebook.github.io/react-native/docs/getting-started.html) 

## Learn React

[React Native Express](http://www.reactnativeexpress.com/) is a great site to get you started, specifically:

- [Get your head around ES6](http://www.reactnativeexpress.com/es6)
- [What is JSX?](http://www.reactnativeexpress.com/jsx)
- [What are Components?](http://www.reactnativeexpress.com/components)
- [React State](http://www.reactnativeexpress.com/data_component_state)
- [Redux](http://www.reactnativeexpress.com/redux)

Once you've got your head around the basics, checkout the [React Native](https://facebook.github.io/react-native/) website, specifically:

- Go through ['The Basics'](https://facebook.github.io/react-native/docs/props.html)
- Gain an understanding of the [components](https://facebook.github.io/react-native/docs/activityindicator.html) React Native provides out of the box
