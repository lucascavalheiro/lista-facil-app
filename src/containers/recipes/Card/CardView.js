/**
 * Recipe View Screen
 *  - The individual recipe screen
 *
 * React Native Starter App
 * https://github.com/mcnamee/react-native-starter-app
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  View,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';
import { Icon } from 'react-native-elements';

// Consts and Libs
import { AppStyles } from '@theme/';

// Components
import { Card, Text } from '@ui/';

/* Styles ==================================================================== */
const styles = StyleSheet.create({
  favourite: {
    position: 'absolute',
    top: -45,
    right: 0,
  },
});

/* Component ==================================================================== */
class RecipeCard extends Component {
  static componentName = 'RecipeCard';

  static propTypes = {
    image: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    body: PropTypes.string.isRequired,
    onPress: PropTypes.func,
    onPressFavourite: PropTypes.func,
    isFavourite: PropTypes.bool,
  }

  static defaultProps = {
    onPress: null,
    onPressFavourite: null,
    isFavourite: null,
  }

  render = () => {
    const { title, body, image, onPress, onPressFavourite, isFavourite } = this.props;

    return (
      <TouchableOpacity activeOpacity={0.8} onPress={onPress}>
        <Card image={image && { uri: image }}>
          <View style={[AppStyles.paddingBottomSml]}>
            <Text h3>{title}</Text>
            <Text>{body}</Text>

            {!!onPressFavourite &&
              <TouchableOpacity
                activeOpacity={0.8}
                onPress={onPressFavourite}
                style={[styles.favourite]}
              >
                <Icon
                  raised
                  name={'star-border'}
                  color={isFavourite ? '#FFFFFF' : '#FDC12D'}
                  containerStyle={{
                    backgroundColor: isFavourite ? '#FDC12D' : '#FFFFFF',
                  }}
                />
              </TouchableOpacity>
            }
          </View>
        </Card>
      </TouchableOpacity>
    );
  }
}

/* Export Component ==================================================================== */
export default RecipeCard;
