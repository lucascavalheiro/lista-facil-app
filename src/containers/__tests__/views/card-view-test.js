/**
 * Test to check if the component renders correctly
 */
/* global it expect */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import CardView from '@containers/recipes/Card/CardView';

it('CardView renders correctly', () => {
  const tree = renderer.create(
    <CardView
      image={'http://placehold.it/300x100'}
      title={'A Cake'}
      content={'It\'s so delicious!'}
    />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
