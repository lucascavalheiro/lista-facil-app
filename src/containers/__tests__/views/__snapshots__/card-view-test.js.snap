// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardView renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#e1e8ee",
          "borderWidth": 1,
          "margin": 15,
          "marginBottom": 0,
          "padding": 15,
          "shadowColor": "rgba(0,0,0, .2)",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        Object {
          "padding": 0,
        },
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#D0D1D5",
            "borderRadius": 2,
          },
          Array [],
        ],
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Image
          resizeMode="cover"
          source={
            Object {
              "uri": "http://placehold.it/300x100",
            }
          }
          style={
            Array [
              Object {
                "flex": 1,
                "height": 150,
                "width": null,
              },
              undefined,
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "padding": 10,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "paddingBottom": 10,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              h1={false}
              h2={false}
              h3={true}
              h4={false}
              h5={false}
              onPress={null}
              p={false}
              style={
                Array [
                  Object {
                    "color": "#0E4EF8",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 17.5,
                    "fontWeight": "500",
                    "left": 0,
                    "lineHeight": 23,
                    "margin": 0,
                    "marginBottom": 4,
                    "right": 0,
                  },
                ]
              }
            >
              A Cake
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              h1={false}
              h2={false}
              h3={false}
              h4={false}
              h5={false}
              onPress={null}
              p={false}
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "lineHeight": 18,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
